tests/check_sssd_files_provider.yml

```diff
@@ -0,0 +1,22 @@
+- name: Check with-files-domain feature exists
+  command: authselect list-features sssd
+  register: __tlog_authselect_features
+  changed_when: false
+
+- name: Check if files domain is currently enabled
+  command: authselect current
+  register: __tlog_authselect_current
+  changed_when: false
+  failed_when: __tlog_authselect_current.rc not in [0, 2]
+
+- name: Read nsswitch.conf
+  set_fact:
+    __nsswitch_contents: "{{ lookup('file', '/etc/nsswitch.conf') }}"
```

Comment:
This will get the contents of /etc/nsswitch.conf on the *controller*, not the *managed* host.  Is this what you want to do?  https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_lookup.html#ansible-collections-ansible-builtin-file-lookup

If you want /etc/nsswitch.conf from the managed host, use `slurp`: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/slurp_module.html#ansible-collections-ansible-builtin-slurp-module

```
- name: Read nsswitch.conf
  slurp:
    src: /etc/nsswitch.conf
    register: __nsswitch_contents


- name: Check if files domain enabled and nsswitch set correctly
  assert:
    that:
      - __nsswitch_contents.contents | b64decode | regex_search('^passwd:\\s+sss', multiline=True)
      - '"with-files-domain" in __tlog_authselect_current.stdout'
  when:
    - '"with-files-domain" in __tlog_authselect_features.stdout'
```

Reply:
No, running on the controller wasn't the intention.  I switched to using slurp as suggested.